diff --git a/src/ex_cmds.c b/src/ex_cmds.c
index 6a1715a6b..370964ac7 100644
--- a/src/ex_cmds.c
+++ b/src/ex_cmds.c
@@ -14,6 +14,8 @@
 #include "vim.h"
 #include "version.h"
 
+#include "dio.h"
+
 #include <float.h>
 
 static int linelen(int *has_tab);
@@ -2885,6 +2887,10 @@ do_ecmd(
 	}
     }
 
+#ifdef __MVS__
+  if (is_dataset(sfname))
+    ffname = sfname;
+#endif
     /*
      * If the file was changed we may not be allowed to abandon it:
      * - if we are going to re-edit the same file
@@ -2903,7 +2909,6 @@ do_ecmd(
 	goto theend;
     }
 
-
     // End Visual mode before switching to another buffer, so the text can be
     // copied into the GUI selection buffer.
     // Careful: may trigger ModeChanged() autocommand
