diff --git a/src/structs.h b/src/structs.h
index b112c6d18..fc72b531f 100644
--- a/src/structs.h
+++ b/src/structs.h
@@ -3254,7 +3254,6 @@ struct file_buffer
 				// b_ffname
     char_u	*b_fname;	// current file name, points to b_ffname or
 				// b_sfname
-
 #ifdef UNIX
     bool	b_dev_valid;	// true when b_dev has a valid number
     dev_t	b_dev;		// device number
@@ -3736,6 +3735,13 @@ struct file_buffer
 #ifdef FEAT_DIFF
     bool	b_diff_failed;	// internal diff failed for this buffer
 #endif
+
+#ifdef __MVS__
+    int needs_conversion;
+    int is_dataset;
+    int reclen;
+    char *dsname;
+#endif
 }; // file_buffer
 
 
